### ISSUE TITLE
[A11y] Search Results - Grid - Search Refinement dropdowns (accessible name)

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#2158]
### Operating System

Windows

### Browser

Firefox

### Affected page

Search

### Admin user?

No

### Issue Description

(This is sort of a sub-ticket to #1928 so we can close that out since most of it is done)

- Visit https://staging.oregondigital.org/catalog?&q%5B%5D=test&search_field=all_fields&utf8=%E2%9C%93&view=gallery
- Look at the "SORT" and "SHOW" buttons in an accessibility inspector, with a screen reader, or just viewing the source
- The labels are great *except* you have no idea what the current value is. Sighted users get this.

I don't know how hard a failure this is, but the experience is definitely different for sighted users - they have more information. If it were just results per page I think it wouldn't be a huge deal since that can be determined in other ways, but having to try to remember which sort a person chose... that could be tricky.

### Suggested Fix

I think the best option is putting the current value to the button's accessible description (**not** label) via something like `aria-description="selected: blah"`, e.g.:

`<button type="button" id="search-sort-button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-controls="search-sort-list" aria-expanded="false" aria-haspopup="true" aria-label="Sort by" aria-description="selected: relevance">
      relevance <span class="caret-dropdown"></span>
</button>`

Otherwise, adding `aria-current` to the selected menu item seems to be a sufficient fix, e.g.:

`<a role="menuitem" tabindex="-1" href="/catalog?locale=en&amp;q%5B%5D=test&amp;search_field=all_fields&amp;sort=score+desc%2C+system_create_dtsi+desc&amp;view=gallery" aria-current="true">relevance</a>`

### Can anybody QA or does this need an expert?

Needs expert

### QA Steps

This isn't impossible to test by a novice, but it is WAY faster to verify with a screen reader than manually checking HTML.

With a screenreader, just tab to the button and verify that it tells you the current value after it's described the button.

If you had to check by hand, you'd need to use something like Firefox's accessibility inspector, and verify that the button has a description associated with it. You have to right-click the button, choose "Inspect Accessibility Properties", then look over the properties list to see if there's a description separately from the name, and that if you read the bits in order (name, then role, then description), you know what's going on.

**Comment by azle** [comment]
Updated  12/26/2024 FIXED

MAC VoiceOver reads the sort and show with more description and label of each drop down.
"Showing 1-20 of 95 items Sort results by..."relevance" and show number of results per page..."20""
